src/webots/control/WbControlledWorld.cpp

```diff
@@ -220,8 +220,16 @@ void WbControlledWorld::addControllerConnection() {
       delete[] buffer;
     }
     foreach (WbRobot *const robot, robots()) {
-      if (robot->isControllerStarted())
-        continue;
+      if (robot->isControllerStarted()) {
+        // if a specific robot have been targeted by giving a name, or only one robot is in the scene (hence it's the target)
+        if ((robotName == robot->name() || robots().size() == 1) && robot->controllerName() == "<extern>") {
+          // automatically disconnect from previous extern controller that may have failed, the slot could be immedialy used to restart
```

Comment:
```suggestion
          // automatically disconnect from previous extern controller that may have failed, the slot could be immediatly used to restart
```

Reply:
And what if there is only one external controller, and the name of the robot doesn't match ? Should not happen often, cause in this case user might better not provide any name, but still...

Reply:
I've still made the change, that a nice idea

Reply:
> And what if there is only one external controller, and the name of the robot doesn't match ? Should not happen often, because in this case user might better not provide any name, but still...

It should work exactly as for the first start: if the `robotName` is set but differs from `robot->name()` it is not disconnected.

Reply:
BTW, I've tried to used this sort of construct, which should appropriate : 
`int nbExternalController = std::count_if(robots().begin(), robots().end(), [](WbRobot *const r) { return r->controllerName() == "<extern>"; });`
But it actually cause segfault :/ not sure what happened, I'm not use to C++ that much...